src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -171,17 +190,19 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 
     register_cli_argument(scope, 'os_disk_name', help='The name of the new VM OS disk.', arg_group='Storage')
     register_cli_argument(scope, 'os_type', help='Type of OS installed on a custom VHD. Do not use when specifiying an URN or URN alias.', arg_group='Storage', **enum_choice_list(['windows', 'linux']))
-    register_cli_argument(scope, 'storage_account', help='The name to use when creating a new storage account or referencing an existing one. If omitted, an appropriate storage account in the same resource group and location will be used, or a new one will be created.', arg_group='Storage')
+    register_cli_argument(scope, 'storage_account', help="Only applicable when use with '--use-unmanaged-disk'. The name to use when creating a new storage account or referencing an existing one. If omitted, an appropriate storage account in the same resource group and location will be used, or a new one will be created.", arg_group='Storage')
     register_cli_argument(scope, 'storage_caching', help='Storage caching type for the VM OS disk', arg_group='Storage', **enum_choice_list(['ReadWrite', 'ReadOnly']))
-    register_cli_argument(scope, 'storage_sku', help='The storage SKU to use for new storage accounts.', arg_group='Storage', **enum_choice_list(SkuName))
-    register_cli_argument(scope, 'storage_container_name', help='Name of the storage container for the VM OS disk.', arg_group='Storage')
+    register_cli_argument(scope, 'storage_sku', help="The storage SKU to use for new storage accounts. Allowed value: Premium_LRS, Standard_LRS. With use with '--use-unmanaged-disk', 3 more options are available: Standard_GRS, Standard_RAGRS, Standard_ZRS. ", arg_group='Storage')
```

Comment:
If the service gives an adequate error when a disallowed SKU is provided, I would argue letting it do that. Or you could reword this to keep the enum choice list, but add text to say "For managed-disks, only Premium_LRS and Standard_LRS are permitted.". Naturally have a validator check that as part of the storage profile resolution.